snap/message.go

```diff
@@ -27,8 +27,26 @@ import (
 // Message contains the ReadCloser field for handling large snapshot. This avoid
 // copying the entire snapshot into a byte array, which consumes a lot of memory.
 //
-// User of Message should close the ReadCloser after sending it.
+// User of Message should close the Message after sending it.
 type Message struct {
 	raftpb.Message
 	ReadCloser io.ReadCloser
+	Donec      chan bool
```

Comment:
`Donec` should probably be `donec` since I assume only `SucceededAndClose` and `FailedAndClose` should write to it. I think that means there needs to be a `snap.NewMessage` to setup the channel, then?